engine/security/src/main/scala/pl/touk/nussknacker/ui/security/api/AuthenticationResources.scala

```diff
@@ -1,28 +1,33 @@
 package pl.touk.nussknacker.ui.security.api
 
+import akka.http.scaladsl.marshalling.ToResponseMarshallable
+import akka.http.scaladsl.model.StatusCodes
+import akka.http.scaladsl.server.directives.AuthenticationDirective
 import akka.http.scaladsl.server.{Directives, Route}
-import akka.http.scaladsl.server.directives.{AuthenticationDirective, RouteDirectives}
 import com.typesafe.config.Config
+import pl.touk.nussknacker.ui.security.api.AuthenticationResources.LoggedUserAuth
 import sttp.client.{NothingT, SttpBackend}
 
 import scala.concurrent.{ExecutionContext, Future}
 
-trait AuthenticationResources {
-  val directive: AuthenticationDirective[LoggedUser]
-  val route: Route = Directives.reject
-  val config: AuthenticationConfiguration
+trait AuthenticationResources extends Directives {
+  val name: String
+  val frontendSettings: ToResponseMarshallable = StatusCodes.NoContent
+
+  def authenticate(): LoggedUserAuth
+
+  final lazy val routeWithPathPrefix: Route =
+    pathPrefix("authentication" / name.toLowerCase() ) {
+      route ~ frontendSettingsRoute
+    }
+
+  protected lazy val frontendSettingsRoute: Route = path("settings") { get { complete { frontendSettings } } }
+  protected lazy val route: Route = Directives.reject
```

Comment:
let's change name to sth more descriptive? additionalRoute? Or add scaladoc that this is the method to override?

Reply:
Changed